api/provider/aws/environment_link.go

```diff
@@ -0,0 +1,84 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+func (e *EnvironmentProvider) Link(req models.CreateEnvironmentLinkRequest) error {
+	fqSourceEnvID := addLayer0Prefix(e.Config.Instance(), req.SourceEnvironmentID)
+	fqDestEnvID := addLayer0Prefix(e.Config.Instance(), req.DestEnvironmentID)
+
+	sourceGroup, err := readSG(e.AWS.EC2, getEnvironmentSGName(fqSourceEnvID))
+	if err != nil {
+		return err
+	}
+
+	destGroup, err := readSG(e.AWS.EC2, getEnvironmentSGName(fqDestEnvID))
+	if err != nil {
+		return err
+	}
+
+	sourceGroupID := sourceGroup.GroupId
+	destGroupID := destGroup.GroupId
+
+	if err := e.createIngressInput(sourceGroupID, destGroupID); err != nil {
+		return err
+	}
+
+	if err := e.createIngressInput(destGroupID, sourceGroupID); err != nil {
+		return err
+	}
+
+	if err := e.createLinkTags(req.SourceEnvironmentID, req.DestEnvironmentID); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (e *EnvironmentProvider) createIngressInput(sourceGroupID, destGroupID *string) error {
```

Comment:
[nit] these args should be `string` instead of `*string` to match your delete helper and ROT